lib/util/Map.v3

```diff
@@ -110,6 +110,50 @@ class HashMap<K, V> extends PartialMap<K, V> {
 			}
 		}
 	}
+	// Remove {key} from this map. Return true if {key} existed.
+	def remove(key: K) -> bool {
+		var c = cache;
+		if (c == null) return false; // empty map
+		var found = false;
+
+		// search table
+		if (table != null) {
+			for (i < table.length) {
+				if (table[i] == null) continue;
```

Comment:
I think you can pull the `b` variable up to here; it's the head of the bucket list.